Automattic/wp-desktop
# Fix Canary Test: Replace spectron with electron-mocha

### Description:
Fix for the calypso e2e canary test. 
With this PR I'm falling back to electron-mocha which previously was working quite well for the automated test.  

**Note:** This PR pretty much restores the test we previously had. 

### Motivation & Context:
During the modernization of most of our build scripts, I've replaced electron-mocha with spectron as we were able to test the app cross process. Unfortunately, with the latest spell checker update, I broke the spectron based canary test which however seems to be highly unreliable in combination with native dependencies. Even on my local machine 1 out of 5 tests failed in a vanilla state. 

### How to test:
- Add `CALYPSO_HASH` env var in the CircleCI config here: https://github.com/Automattic/wp-desktop/blob/develop/.circleci/config.yml#L87
```yml
environment: 
  CALYPSO_HASH: c5d4220f0cc3114467c4b1bc2e8155ef4d20233f
```
- Push your changes and wait for CircleCI to turn green

_or_ 

- Check the `test` step here: 
https://circleci.com/gh/Automattic/wp-desktop/21504

### Take aways:
- We need a better testing strategy with regular CI builds (not just calypso triggered e2e tests)
- We shouldn't rely on ipc – ping-pong based testing for scaling tests

I'm going to prepare a P2 post with more details.
